### DISCUSSION TITLE
How to tag a new version of the dataset?

### DISCUSSION THREAD

**Discussion by msis** [huggingface/datasets#5370] [Q&A]
Currently, when uploading a dataset, one would call:
```py
dataset.push_to_hub(
  "user_name/dataset_name",
)
```

So I thought one could create a new version with:
```py
dataset.push_to_hub(
  "user_name/dataset_name",
  branch="v1",
)
```

But then I get the following error:
```
Revision Not Found for url: https://huggingface.co/api/datasets/tarteel-ai/tldn_hf/revision/v1.                                                                                                            
Invalid rev id: v1  
```

How to then create a new version?

I've tried cloning, tagging the last commit, pushing it, then calling `push_to_hub` again.

However, I seed now duplicates of shard, see below:

<img width="348" alt="Screenshot 2022-12-17 at 12 50 17" src="https://user-images.githubusercontent.com/577139/208240408-eeae7132-f799-466c-bc0b-e5b746c17f12.png">

How do I then get only one version in the commit of `v1`?

**Comment by mariosasko** [comment]
Hi! You can use Git tags to mark the releases (versions) of a dataset:
```python
import huggingface_hub

# version 1.0
dataset.push_to_hub(
  "user_name/dataset_name",
)
huggingface_hub.create_tag("user_name/dataset_name", tag="1.0", repo_type="dataset")

# version 2.0
dataset.push_to_hub(
  "user_name/dataset_name",
)
huggingface_hub.create_tag("user_name/dataset_name", tag="2.0", repo_type="dataset")
```
And then reference them when loading as follows:
```python
# version 1.0
load_dataset("user_name/dataset_name", revision="1.0")
# version 2.0
load_dataset("user_name/dataset_name", revision="2.0")
```

PS: `push_to_hub` can only push to an existing branch. So if `branch` points to a non-existing branch, call `huggingface_hub.create_branch("user_name/dataset_name", branch=branch, repo_type="dataset")` before running `push_to_hub`.

**Reply by xhluca** [reply]
I was looking for this information and finally found it. This worked great, thanks! It would be amazing if it would be added here: https://huggingface.co/docs/datasets/v3.5.0/en/share#share-a-dataset-using-the-cli
